lib/lintSource.js

```diff
@@ -88,7 +87,7 @@ function lintPostcssResult(
   postcssResult.stylelint.quiet = config.quiet
 
   const newlineMatch = postcssResult.root.toResult().css.match(/\r?\n/)
-  const newline = newlineMatch ? newlineMatch[0] : os.EOL
+  const newline = newlineMatch ? newlineMatch[0] : "\n"
```

Comment:
I wish it works :(

With:

```js
const os = require("os")
jest.mock("os", () => Object.assign({}, os, { EOL: "\n" }))
```

Error:

```
● Test suite failed to run

  /Users/aleks/projects/stylelint/jest-setup.js: babel-plugin-jest-hoist: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.
  Invalid variable access: os
  Whitelisted objects: Array, ArrayBuffer, Boolean, DataView, Date, Error, EvalError, Float32Array, Float64Array, Function, Generator, GeneratorFunction, Infinity, Int16Array, Int32Array, Int8Array, InternalError, Intl, JSON, Map, Math, NaN, Number, Object, Promise, Proxy, RangeError, ReferenceError, Reflect, RegExp, Set, String, Symbol, SyntaxError, TypeError, URIError, Uint16Array, Uint32Array, Uint8Array, Uint8ClampedArray, WeakMap, WeakSet, arguments, expect, jest, require, undefined, DTRACE_NET_SERVER_CONNECTION, DTRACE_NET_STREAM_END, DTRACE_HTTP_SERVER_REQUEST, DTRACE_HTTP_SERVER_RESPONSE, DTRACE_HTTP_CLIENT_REQUEST, DTRACE_HTTP_CLIENT_RESPONSE, global, process, Buffer, clearImmediate, clearInterval, clearTimeout, setImmediate, setInterval, setTimeout, console.
  Note: This is a precaution to guard against uninitialized mock variables. If it is ensured that the mock is required lazily, variable names prefixed with `mock` are permitted.

    at invariant (node_modules/babel-plugin-jest-hoist/build/index.js:13:11)
    at newFn (node_modules/babel-traverse/lib/visitors.js:276:21)
    at NodePath._call (node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitMultiple (node_modules/babel-traverse/lib/context.js:103:17)
```

---

With:

```js
const mockos = require("os")
jest.mock("os", () => Object.assign({}, mockos, { EOL: "\n" }))
```

Error:

```
● Test suite failed to run

  ReferenceError: mockos is not defined

    at Object.<anonymous>.jest.mock [as node:os:] (jest-setup.js:141:41)
    at Object.<anonymous> (node_modules/os-homedir/index.js:2:10)
    at Object.<anonymous> (node_modules/cosmiconfig/index.js:4:17)
```

---

With:

```js 
jest.mock("os", () => {
  const nodeos = require("os")

  return Object.assign({}, nodeos, { EOL: "\n" })
})
```

Error:

```
● Test suite failed to run

  RangeError: Maximum call stack size exceeded

    at Resolver.getModulePath (node_modules/jest-resolve/build/index.js:185:16)
    at Object.<anonymous>.jest.mock [as node:os:] (jest-setup.js:141:18)
    at Object.<anonymous>.jest.mock [as node:os:] (jest-setup.js:141:18)
    at Object.<anonymous>.jest.mock [as node:os:] (jest-setup.js:141:18)
```